lib/editor/util/gtfs.js

```diff
@@ -220,7 +227,47 @@ export function getEntityBounds (entity: any, stops?: Array<GtfsStop>, offset: n
             ...pattern.shape.coordinates.map(c => [c[1], c[0]])
           ]
         }
-      })
+
+        let flexRouteCoordinates = []
+
+        // check for flex route locations
+
+        if (locations && pattern.patternLocations && pattern.patternLocations.length > 0) {
+          // finds matching locations and coordinates in state + concats all locations into one array of coordinate objects
+
+          const locationCoordinates = pattern.patternLocations
+            .map(pl => locations ? locations.find(l => l.location_id === pl.locationId) : null)
+            .map(pl => pl ? pl.location_shapes : null)
+            // $FlowFixMe flow doesn't understand this but it works
+            .flat(1)
+
+          flexRouteCoordinates = [
+            ...flexRouteCoordinates,
+            ...locationCoordinates
+          ]
+        }
+
+        // check for flex trip pattern stops, and add to flex coordinates array
+
+        if (stops && pattern.patternStops && pattern.patternStops.length > 0) {
+          flexRouteCoordinates = [
+            ...flexRouteCoordinates,
+            ...stops
+          ]
+        }
```

Comment:
I think this could be compressed into a single declaration -- both the locations and the stops could be items in a new array and if wrapped in `()` you could add a `|| []` at the end to avoid the null errors.

I realize that's not the best explanation for what I'm trying to describe. We can chat more outside of github

Reply:
I was able to make this much shorter, but the `if` condition may be a little long + convoluted so I'm very open to feedback!!

Also may have missed the mark big time, just let me know

Reply:
Looks great!